sys-boot/udk/files/udk-workspace.template

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+show_usage() {
+	if [ ! -z "${1}" ]; then
+		echo "${1}" >&2
+		echo >&2
+	fi
+	echo "Usage:"
+	echo ". ${BASH_SOURCE[0]} [folder]"
+	echo "where folder is the workspace folder - current directory is used if unspecified"
+}
+
+create_workspace() {
+	local workspace_dir="${PWD}"
+	if [ ! -z "${1}" ]; then
+		workspace_dir="${1}"
+		if [ ! -d "${1}" ]; then
+			if ! mkdir -p "${1}"; then
+				show_usage "Cannot create directory ${1}"
+				return 1
+			fi
+		fi
+	fi
+
+	pushd "${workspace_dir}" >>/dev/null
+	if [ ! -d Conf ] && ! mkdir Conf; then
+		show_usage "Cannot create configuration directory"
+		return 1
+	fi
+	export WORKSPACE="${PWD}"
+	popd >>/dev/null
+}
+
+if [ "${1}" = "-h" ] || [ "${1}" = "--help" ]; then
```

Comment:
Unless I'm mistaken, this can match `${1}` of sourcing script.

Reply:
Correct… :disappointed: 
Should I only accept these parameters on direct calls?

Reply:
I think I'll keep this and hope users will clear parameters before calling the script. For almost every usage, this should work correctly. It may fail if the user source the script inside a script taking parameters, and in this case, he will have to take care about those parameters.